### PR TITLE
Bugfix/details screen UI

### PR DESCRIPTION
**Pull request by sjacobs1**
##Description
changed Movie details UI to look better, rounded corners and movie poster size update.
Add button now changes when movie is added and shows if movie is saved or not.
code cleanup (moved code in extension to main class)

##GIF
ezgif com-optimize-3

#Checklist
- [x]  The code follows the project's coding standards.
- [ ]  100% Unit test coverage for Tested View Models
- [x]  All warnings Resolved.

### CODE REVIEW

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift [hunk 1]
```diff
@@ -32,18 +33,16 @@ class MovieDetailsViewController: UIViewController {
...
-}
 
-extension MovieDetailsViewController: MovieDetailsViewModelType {
     func didUpdateMovieDetails() {
-        DispatchQueue.main.async {
-            self.updateUI()
-        }
+        self.updateUI()
+        self.updateWatchlistButton()
     }
```
Comment (sjacobs1) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift:46]: make this function name more descriptive : "updateWatchlistButtonTitle"

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift [hunk 2]
```diff
@@ -32,18 +33,16 @@ class MovieDetailsViewController: UIViewController {
...
         movieDetailsViewModel.updateMovieID(movieID: movieID)
     }
-}
 
-extension MovieDetailsViewController: MovieDetailsViewModelType {
     func didUpdateMovieDetails() {
-        DispatchQueue.main.async {
-            self.updateUI()
-        }
+        self.updateUI()
```
Comment (sjacobs1) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift:44]: function can be named better here too

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift [hunk 3]
```diff
@@ -32,18 +33,16 @@ class MovieDetailsViewController: UIViewController {
...
+        updateWatchlistButton()
     }
 
     func setMovieID(movieID: Int) {
         movieDetailsViewModel.updateMovieID(movieID: movieID)
     }
-}
 
-extension MovieDetailsViewController: MovieDetailsViewModelType {
     func didUpdateMovieDetails() {
```
Comment (sjacobs1) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift]: this function can also be named better : "updateMovieDetailsUI"

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift [hunk 4]
```diff
@@ -52,10 +51,17 @@ extension MovieDetailsViewController: MovieDetailsViewModelType {
...
+        if movieDetailsViewModel.isMovieSaved {
+            addToWatchlistButton.setTitle("✓ Saved", for: .normal)
+        }
+    }
+
     private func updateUI() {
         titleLabel.text = movieDetailsViewModel.originalTitle
         overviewLabel?.text = movieDetailsViewModel.overview
-        releaseDateLabel.text = movieDetailsViewModel.releaseDate
+        releaseDateLabel.text = String(movieDetailsViewModel.releaseDate?.prefix(4) ?? "")
```
Comment (kieran-woodrow) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift]: rather don't cast this here, let the function return a string

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift (hunk 4 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsViewController.swift]: understood. change made

**Review by kieran-woodrow — APPROVED** [review]

**Review by kiasha10 — APPROVED** [review]

**Pull request merged**